tests/lib/eslint.js

```diff
@@ -3189,6 +3189,32 @@ describe("eslint", function() {
             eslint.verify("var { a='a' } = {};", { parserOptions: { ecmaVersion: 6 }});
         });
 
+        it("should not rewrite env setting in core (https://github.com/eslint/eslint/issues/4814)", function() {
+            eslint.defineRule("test", function() {
+                return {};
+            });
+            eslint.verify("var a = 0;", {
+                env: {node: true},
```

Comment:
Why are you calling `verify` twice? I'm unclear on what the actual error is from looking at this test.